src/pin_vec.rs

```diff
@@ -22,6 +23,12 @@ impl<T> PinVec<T> {
     }
 
     pub fn clear(&mut self) {
```

Comment:
Two suggestions:
* Each bucket could be iterated over. Since it has a known length could be reconstituted as a slice up until its initialized range and dropped more efficiently (see [Vec's Drop impl](https://doc.rust-lang.org/src/alloc/vec/mod.rs.html#2821)).
* We can avoid the loop if `std::mem::needs_drop::<T>()` is false.

I don't think we need to deal with ZSTs, since we never store any. PinSlab uses `Entry<T>` which is non-zero even if `T` is a ZST. `InternalWaker` isn't zero-sized.

Reply:
Good suggestion. It makes the code more complicated, but I think your suggestion is better so I went ahead and applied it.

Reply:
Since now we're dropping through a slice, hopefully that handles any difficulty ZSTs might give us?

Reply:
Yeah, it should be good. Cheers!